src/CoreWCF.NetTcp/tests/ServiceAuthBehaviorTest.cs

```diff
@@ -128,14 +128,25 @@ private void assertForCommon(String sourceString,  IWebHost host)
                     factory = new System.ServiceModel.ChannelFactory<ClientContract.ITestService>(binding,
                         new System.ServiceModel.EndpointAddress(new Uri(WindowsAuthNetTcpServiceUri)));
                     channel = factory.CreateChannel();
+                    
+                    // ReSharper disable once SuspiciousTypeConversion.Global
                     ((IChannel)channel).Open();
+
                     var result = channel.EchoForPermission(sourceString);
                     Assert.Equal(sourceString, result);
-                    ((IChannel)channel).Close();
-                    factory.Close();
+
+                    //
+                    // These were explicitly removed because the ServiceHelper already cleans these up, and
+                    // in some cases not using proper disposal handling will result in:
+                    // 'System.IO.IOException : Received an unexpected EOF or 0 bytes from the transport stream.'
+                    // on the build server, causing false test failures.
+                    //
+                    // ((IChannel)channel).Close();
```

Comment:
The intention of having these as part of the test code is to cause an exception to be thrown if there's a problem with cleanup. For example, there's a close handshake for NetTcp which if the server isn't correctly handling could cause an exception to be thrown here and we want to know about that. The code in ServiceHelper.CloseServiceModelObjects is designed to suppress any exceptions as it's purpose is to prevent any resource leaks such as the channel closing throws so the factory doesn't get closed which leaves sockets connected in an idle connection pool.  
Do you know what exception was being thrown here?

Reply:
It's in the comment: 

> These were explicitly removed because the ServiceHelper already cleans these up, and
> in some cases not using proper disposal handling will result in:
> 'System.IO.IOException : Received an unexpected EOF or 0 bytes from the transport stream.'
> on the build server, causing false test failures.

Reply:
This is suggesting there's a bug in the server implementation code on the close code path. This change is just hiding that. I'll try and see if I can work out what's causing this.